src/Microsoft.AspNetCore.Rewrite/IISUrlRewriteOptionsExtensions.cs

```diff
@@ -35,16 +38,18 @@ public static RewriteOptions AddIISUrlRewrite(this RewriteOptions options, IFile
 
             using (var stream = file.CreateReadStream())
             {
-                return AddIISUrlRewrite(options, new StreamReader(stream));
+                return AddIISUrlRewrite(options, new StreamReader(stream), rewriteMaps);
             }
         }
 
-        /// <summary>
-        /// Add rules from a IIS config file containing Url Rewrite rules
-        /// </summary>
-        /// <param name="options">The <see cref="RewriteOptions"/></param>
-        /// <param name="reader">The text reader stream.</param>
-        public static RewriteOptions AddIISUrlRewrite(this RewriteOptions options, TextReader reader)
+		/// <summary>
+		/// Add rules from a IIS config file containing Url Rewrite rules
+		/// </summary>
+		/// <param name="options">The <see cref="RewriteOptions"/></param>
+		/// <param name="reader">The text reader stream.</param>
+		/// <param name="rewriteMaps">An optional set of rewrite maps to uztilize when parsing rules.
+		/// Rewrite maps in this collection will overwrite (supercede) duplicate named entries in the XML.</param>
+		public static RewriteOptions AddIISUrlRewrite(this RewriteOptions options, TextReader reader, IEnumerable<IISRewriteMap> rewriteMaps = null)
```

Comment:
Is that a supported IIS scenario? We're not trying to add functionality beyond IIS at this point.